tensornetwork/block_sparse/charge.py

```diff
@@ -114,3 +114,272 @@ def __repr__(self):
+  #pylint: disable=no-else-return
+  if A.ndim == 1:
+    return np.intersect1d(
+        A, B, assume_unique=assume_unique, return_indices=return_indices)
+
+  elif A.ndim == 2:
+    if axis == 0:
+      ncols = A.shape[1]
+      if A.shape[1] != B.shape[1]:
+        raise ValueError("array widths must match to intersect")
+
+      dtype = {
+          'names': ['f{}'.format(i) for i in range(ncols)],
+          'formats': ncols * [A.dtype]
+      }
+      if return_indices:
+        C, A_locs, B_locs = np.intersect1d(
+            A.view(dtype),
+            B.view(dtype),
+            assume_unique=assume_unique,
+            return_indices=return_indices)
+        return C.view(A.dtype).reshape(-1, ncols), A_locs, B_locs
+      C = np.intersect1d(
+          A.view(dtype), B.view(dtype), assume_unique=assume_unique)
+      return C.view(A.dtype).reshape(-1, ncols)
+
+    elif axis == 1:
+      #@Glen: why the copy here?
+      out = intersect(
+          A.T.copy(),
+          B.T.copy(),
+          axis=0,
+          assume_unique=assume_unique,
+          return_indices=return_indices)
+      if return_indices:
+        return out[0].T, out[1], out[2]
+      return out.T
+
+    raise NotImplementedError(
+        "intersection can only be performed on first or second axis")
+
+  raise NotImplementedError("intersect is only implemented for 1d or 2d arrays")
+
+
+def fuse_charges(charges: List[BaseCharge], flows: List[bool]) -> BaseCharge:
+  """
+  Fuse all `charges` into a new charge.
+  Charges are fused from "right to left",
+  in accordance with row-major order.
+
+  Args:
+    charges: A list of charges to be fused.
+    flows: A list of flows, one for each element in `charges`.
+  Returns:
+    BaseCharge: The result of fusing `charges`.
+  """
+  if len(charges) != len(flows):
+    raise ValueError(
+        "`charges` and `flows` are of unequal lengths {} != {}".format(
+            len(charges), len(flows)))
+  fused_charges = charges[0] * flows[0]
+  for n in range(1, len(charges)):
+    fused_charges = fused_charges + charges[n] * flows[n]
+  return fused_charges
+
+
+def fuse_degeneracies(degen1: Union[List, np.ndarray],
+                      degen2: Union[List, np.ndarray]) -> np.ndarray:
+  """
+  Fuse degeneracies `degen1` and `degen2` of two leg-charges
+  by simple kronecker product. `degen1` and `degen2` typically belong to two
+  consecutive legs of `BlockSparseTensor`.
+  Given `degen1 = [1, 2, 3]` and `degen2 = [10, 100]`, this returns
+  `[10, 100, 20, 200, 30, 300]`.
+  When using row-major ordering of indices in `BlockSparseTensor`,
+  the position of `degen1` should be "to the left" of the position of `degen2`.
+  Args:
+    degen1: Iterable of integers
+    degen2: Iterable of integers
+  Returns:
+    np.ndarray: The result of fusing `dege1` with `degen2`.
+  """
+  return np.reshape(degen1[:, None] * degen2[None, :],
```

Comment:
This slicing trick won't work with a python list I believe.

Reply:
absolutely not! I'll fix it